node/metered-channel/src/lib.rs

```diff
@@ -108,10 +107,16 @@ impl Meter {
 /// Determine if this instance shall be measured
 #[inline(always)]
 fn measure_tof_check(nth: usize) -> bool {
-	let mut rng = rand::thread_rng();
-	let val = rng.gen_range(0..1000);
-	const TOF_MEASURE_EVERY_NTH: usize = if cfg!(test) { 2 } else { 17 };
-	val == 0 && ((TOF_MEASURE_EVERY_NTH - 1_usize) & nth) == 0
+	if cfg!(test) {
+		// for tests, be deterministic and pick every second
+		nth & 0x01 == 0
+	} else {
+		use nanorand::Rng;
+		let mut rng = nanorand::WyRand::new_seed(nth as u64);
```

Comment:
A side effect of this initialisation would be that given a message number we can easily say if it would be sampled or not. So we have a completely deterministic but uniform distribution of samples after all. For a completely random sequence we would need `WyRand::new` (which seeds from the system entropy) wrapped by lazy_static I suppose. However, this deterministic approach might have certain benefits as well.

Reply:
Sampling doesn't require the `csrng`, any `rng` is sufficient, so there is little value in using `lazy_static` which adds another atomic (at the very least).